### ISSUE TITLE
Ideas to improve localization

### ISSUE THREAD

**Issue by daPhipz** [quickgui/quickgui#15]
German (and I am sure some other languages as well) has a sentence structure that's a little different from English. Sentences that are made up of a translatable string and some variable (e.g. a VM name) would need to include a placeholder for the variable in order to be fully localizable. For example, `"Select version for VARIABLE"` would be `"Version für VARIABLE auswählen"` in German. Currently, this is not possible.

I am a member of the German GNOME Translation team, and I come across strings that contain `%s` or `{}` as a placeholder for these variables quite often. I think the first one is used in Python? Maybe there's a Flutter/Dart equivalent...

**Comment by ymauray** [comment]
Yes indeed, that is on my `todo` list! Now that I have another person interested in translating `quickgui`, I'll put that at the top of the priority list!

Thanks for your involvement in this project!

**Comment by daPhipz** [comment]
Nice, looking forward to it!

**Comment by ymauray** [comment]
Fixed in v1.1.4

We can now have placeholders in the messages, as you see below.

You'll have to look at the german file, lines 62 and 63, I'm pretty sure this is wrong, even though I don't speak german ^^
```
msgid "Now run {0} to start the VM"
msgstr "Now run {0} um die VM zu starten"
```

**Comment by daPhipz** [comment]
Looks nice 👍
I'll have a look at it tonight after work!

**Comment by daPhipz** [comment]
Thanks for the updates! I checked the German translation and made some changes, which I'll PR shortly
While testing, I found some issues:
- In the Downloader, the string in front of the working directory is still English.
- The System notification when a Download finished is still in English.
- The translation for `"Downloading... {0}"` doesn't seem to get used, and the target directory is labeled twice while the download is in progress:
![Screenshot_2021-11-14_21:15:58](https://user-images.githubusercontent.com/30795174/141697081-32a76389-fd6f-470f-8ad7-5e1885c76443.png)

**Comment by ymauray** [comment]
I'll look into that !

**Comment by daPhipz** [comment]
OK, I'll wait with my translation update PR then, since it is very small and I can then incorporate the above stuff in it :+1:

**Comment by ymauray** [comment]
All fixed in v1.1.5
